java/bogdana_maksiuta/src/com/zoolatech/lecture3/tasks/_4/Reporter.java

```diff
@@ -0,0 +1,34 @@
+package com.zoolatech.lecture3.tasks._4;
+
+public abstract class Reporter {
+    String headerData;
+    String fileData;
+    String fileName;
+
+    public Reporter(String headerData, String fileData, String fileName) {
+        this.headerData = headerData;
+        this.fileData = fileData;
+        this.fileName = fileName;
+    }
+
+    void generate() {
```

Comment:
Almost correct:
1. Children classes don't need to override method generate - they should used the version of a parent class
2. Methods `addHeaderData` and `addFileData` should be abstract in the parent class, and overridden in children classes